auth/src/main/java/com/firebase/ui/auth/util/smartlock/SignInDelegate.java

```diff
@@ -0,0 +1,403 @@
+import static android.app.Activity.RESULT_FIRST_USER;
+import static android.app.Activity.RESULT_OK;
+
+public class SignInDelegate extends SmartLock<CredentialRequestResult> {
+    private static final String TAG = "SignInDelegate";
+    private static final int RC_CREDENTIALS_READ = 2;
+    private static final int RC_IDP_SIGNIN = 3;
+    private static final int RC_AUTH_METHOD_PICKER = 4;
+    private static final int RC_EMAIL_FLOW = 5;
+    private static final int RC_PLAY_SERVICES = 6;
+
+    private AuthUI.AuthUIResult mAuthUIResult;
+    private FlowParameters mFlowParams;
+    private ProgressDialog mProgressDialog;
+    private GoogleApiClient mGoogleApiClient;
+    private Credential mCredential;
+
+    @Override
+    public void onCreate(Bundle savedInstance) {
+        super.onCreate(savedInstance);
+        setRetainInstance(true);
+        // Make Google Play Services available at the correct version, if possible
+        boolean madeAvailable =
+                PlayServicesHelper
+                        .getInstance(getActivity())
+                        .makePlayServicesAvailable(getActivity(), RC_PLAY_SERVICES,
+                                                   new DialogInterface.OnCancelListener() {
+                                                       @Override
+                                                       public void onCancel(DialogInterface dialogInterface) {
+                                                           Log.w(TAG,
+                                                                 "playServices:dialog.onCancel()");
+                                                           finish(RESULT_CANCELED, new Intent());
+                                                       }
+                                                   });
+
+        if (!madeAvailable
+                || !PlayServicesHelper.getInstance(getActivity()).isPlayServicesAvailable()
+                || !FirebaseAuthWrapperFactory.getFirebaseAuthWrapper(mFlowParams.appName)
+                .isPlayServicesAvailable(getActivity())) {
+            Log.w(TAG, "playServices: could not make available.");
+            finish(RESULT_CANCELED, new Intent());
+            return;
+        }
+
+        if (mFlowParams.smartLockEnabled) {
+            showProgress();
+            initGoogleApiClient(null);
+            Auth.CredentialsApi
+                    .request(mGoogleApiClient,
+                             new CredentialRequest.Builder()
+                                     .setPasswordLoginSupported(true)
+                                     .setAccountTypes(IdentityProviders.GOOGLE)
+                                     .build())
+                    .setResultCallback(this);
+        } else {
+            startAuthMethodChoice();
+        }
+    }
+
+    @Override
+    public void onResult(@NonNull CredentialRequestResult result) {
+        Status status = result.getStatus();
+
+        if (status.isSuccess()) {
```

Comment:
Here's where I removed the checks to see if play services was available and if we care about smart lock.

Also, I simplified the whole isAutoSignInAvailable isSignInResolutionNeeded thing since the only time their values ever changed was in onResult which is the only place this stuff gets called.